ci: mirror-data-to-staging-on-deploy

atolopko-czi opened the pull request:
- run mirroring on staging deploy
- temporarily disable real deploys during testing

## Reason for Change

- #TICKET_NUMBER
- If the reason for this PR's code changes are not clear in the issue, state value/impact

## Changes

- add
- remove
- modify

## Testing steps

- Either list QA steps or reasoning you feel QA is unnecessary
- Describe how you made sure to know that your changes worked. Should allow someone else to go verify your code without in depth knowledge.
- "Unit tested only", "tested in rdev by a, b, c, verifying feature worked by... ", "manually ran pipeline locally with these results: ..."

## Notes for Reviewer

atolopko-czi committed:
run mirroring on staging deploy

runs upstream of post-deployment tests

atolopko-czi committed:
temporarily disable real deploys during testing

atolopko-czi committed:
enable testing on updates to this branch

atolopko-czi committed:
enable testing on updates to this branch

atolopko-czi committed:
GHA data mirroring for staging, individual cmds

The existing scripts are not usable without major changes, due to differences
in how IAM roles are assumed in local usage vs GHA usage.

prathapsridharan reviewed (commented):
Looks good!

1. To be extra safe, might it be worthwhile to take a manual backup of prod data before running this? 
2. And maybe even have a script ready to update `prod` with the backup in case something does go wrong? (maybe this is over-cautious)

atolopko-czi commented:
> Looks good!
> 
> 1. To be extra safe, might it be worthwhile to take a manual backup of prod data before running this?
> 2. And maybe even have a script ready to update `prod` with the backup in case something does go wrong? (maybe this is over-cautious)

I'l do 1 for sure, and I'll create an RDS snapshot as well.

atolopko-czi committed:
let it ride...

atolopko-czi committed:
Remove RDS sync mirroring

This is not a viable design, due to GHA being outside of VPC! Will replace
with a proper design.

atolopko-czi committed:
debug aws s3 sync cmd

atolopko-czi committed:
do not run other workflows during testing

atolopko-czi committed:
install aws-cli v2

atolopko-czi committed:
fix

atolopko-czi committed:
debug

atolopko-czi committed:
fix

atolopko-czi committed:
Draft implementation of ECS-based data mirroring from GHA

atolopko-czi committed:
gha fix
